<a href="https://colab.research.google.com/github/adas754/python_revision-from-day1/blob/main/python_140.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# Program 1
# Write a Python program to print "Hello Python"

In [ ]:
print('Hello Python')

Hello Python


#  Program 2 Write a Python program to do arithmetical operations addition and division.

In [ ]:
def add_div(a,b):
  c=a+b
  d=a/b
  return c,d
add_div(25,5)

(30, 5.0)

# Program 3 Write a Python program to find the area of a triangle.

In [ ]:
def area(h,b):
  return 0.5*h*b
area(10,15)

75.0

# Program 4 Write a Python program to swap two variables.

In [ ]:
def swap(a,b):
  a=a+b
  b=a-b
  a=a-b
  return a,b
a,b=swap(34,67)
print(a,b)

67 34


# Program 5
# Write a Python program to generate a random number.

In [ ]:
import random
random.randint(1,100)

25

# Program 6
# Write a Python program to convert kilometers to miles

In [ ]:
def km_mile(km):
  miles=km* 0.621371
  return miles
km_mile(100)


62.137100000000004

# Program 7
# Write a Python program to convert Celsius to Fahrenheit.

In [ ]:
def cel_far(c):
  f=(c*9/5)+32
  return f
cel_far(37)

98.6

# Program 8
# Write a Python program to display calendar.

In [ ]:
import calendar
year = int(input("Enter year: "))
month = int(input("Enter month: "))
cal = calendar.month(year, month)
print(cal)


Enter year: 2023
Enter month: 11
   November 2023
Mo Tu We Th Fr Sa Su
       1  2  3  4  5
 6  7  8  9 10 11 12
13 14 15 16 17 18 19
20 21 22 23 24 25 26
27 28 29 30



# Program 9
# Write a Python program to solve quadratic equation.

In [ ]:
import cmath
a=int(input())
b=int(input())
c=int(input())
d=(b**2-4*a*c)**0.5
root1=(-b+d)/(2*a)
root2=(-b-d)/(2*a)
print(root1,root2)


1
4
8
(-1.9999999999999998+2j) (-2-2j)


# Program 10
# Write a Python program to swap two variables without temp variable.

In [ ]:
def swap_num(a,b):
  a=a+b
  b=a-b
  a=a-b
  return a,b
swap_num(56,57)

(57, 56)

# Program 11 Write a Python Program to Check if a Number is Positive, Negative or Zero.

In [ ]:
def pos_num(n):
  if n>0:
    print('positive')
  elif n==0:
    print('zero')
  else:
    print('negative')
pos_num(56)


positive


# Program 12 Write a Python Program to Check if a Number is Odd or Even

In [ ]:
def eve_odd(n):
  if n%2==0:
    print('even')
  else:
    print('odd')
eve_odd(31)

odd


# Program 13 Write a Python Program to Check Leap Year.

In [ ]:
def leap_yr(yr):
  if(yr%4==0 and yr%100!=0 or yr%400==0):
    print('year is leap year')
  else:
    print('not leap year')
leap_yr(2024)

year is leap year


# Program 14
# Write a Python Program to Check Prime Number.


In [ ]:
def check_prime(n):
  if n<=1:
    return False
  for i in range(2,int(n**0.5)+1):
    if n%i==0:
      return False
  return True
check_prime(19)

True

# Program 15
# Write a Python Program to Print all Prime Numbers in an Interval of 1-10.


In [ ]:
def is_prime(n):
  if n<=1:
    return False
  for i in range(2,int(n**0.5)+1):
    if n%i==0:
      return False
  return True
def print_prime(start,end):
  for i in range(start,end+1):
    if is_prime(i):
      print(i)
print_prime(1,10)

2
3
5
7


# Program 16
# Write a Python Program to Find the Factorial of a Number.


In [ ]:
def fact_number(n):
  if n==0 or n==1:
    return 1
  else:
    return n*fact_number(n-1)
fact_number(5)

120

# Program 17
# Write a Python Program to Display the multiplication Table.


In [ ]:
n=int(input())
for i in range(1,11):
  print(f'{n}x{i}={n*i}')

10
10x1=10
10x2=20
10x3=30
10x4=40
10x5=50
10x6=60
10x7=70
10x8=80
10x9=90
10x10=100


# Program 18
# Write a Python Program to Print the Fibonacci sequence.

In [ ]:
def fib_num(n):
  if n<=1:
    return n
  else:
    return (fib_num(n-1)+fib_num(n-2))
for i in range(10):
  print(fib_num(i))

0
1
1
2
3
5
8
13
21
34


# Program 19
# Write a Python Program to Check Armstrong Number?

In [ ]:
def arm_num(num):
  num_str=str(num)
  digit_num=len(num_str)
  sum=0
  for digit in num_str:
    sum+=int(digit)**digit_num
  if sum==num:
    return True
  else:
    return False
arm_num(9474)

True

# Program 20
# Write a Python Program to Find Armstrong Number in an Interval.

In [ ]:
def arm_num(num):
  num_str=str(num)
  digit_num=len(num_str)
  sum=0
  for digit in num_str:
    sum+=int(digit)**digit_num
  if sum==num:
    return True
  else:
    return False
def print_arm(start,end):
  for i in range(start,end+1):
    if arm_num(i):
      print(i)
print_arm(10,1000)

153
370
371
407


# Program 21
# Write a Python Program to Find the Sum of Natural Numbers.

In [ ]:
def son(n):
  s=0
  for i in range(1,n+1):
    s+=i
  return s
son(10)

55

# Program 22
# Write a Python Program to Find LCM.

In [ ]:
def compute_lcm(x,y):
  if x>y:
    greater=x
  else:
    greater=y
  while True:
    if((greater%x==0)and(greater%y==0)):
      lcm=greater
      break
    greater+=1
  return lcm
compute_lcm(54,24)

216

# Program 23
# Write a Python Program to Find HCF.


In [ ]:
def compute_hcf(x,y):
  if x>y:
    smaller=y
  else:
    smaller=x
  for i in range(1,smaller+1):
    if((x%i==0)and(y%i==0)):
      hcf=i
  return hcf
compute_hcf(54,24)

6

# Program 24
# Write a Python Program to Convert Decimal to Binary, Octal and Hexadecimal.

In [ ]:
dec_num=int(input("Enter decimal "))
print(bin(dec_num),'in binary')
print(oct(dec_num),'in octal')
print(hex(dec_num),'in hexadecimal')

Enter decimal 27
0b11011 in binary
0o33 in octal
0x1b in hexadecimal


# Program 25
# Write a Python Program To Find ASCII value of a character.


In [ ]:
char=str(input('Enter the character :'))
print('asci value of : ',ord(char))

Enter the character :p
asci value of :  112


# Program 26
# Write a Python Program to Make a Simple Calculator with 4 basic mathematical operations.

In [ ]:
def calculate(a,b,operator):
  if operator=='+':
    return a+b
  elif operator=='-':
    return a-b
  elif operator=='*':
    return a*b
  elif operator=='/':
    return a/b
  else:
    return 'invalid operator'
def main():
  while True:
    a=float(input('Enter 1st number :'))
    b=float(input('Enter 2nd number :'))
    operator=input('Enter operator(+,-,*,/):')
    result=calculate(a,b,operator)
    print(f"result:{result}")
    if input('do you want to continue(y/n) ?')=='n':
      break
main()




Enter 1st number :5
Enter 2nd number :4
Enter operator(+,-,*,/):+
result:9.0
do you want to continue(y/n) ?y
Enter 1st number :5
Enter 2nd number :5
Enter operator(+,-,*,/):-
result:0.0
do you want to continue(y/n) ?y
Enter 1st number :5
Enter 2nd number :4
Enter operator(+,-,*,/):*
result:20.0
do you want to continue(y/n) ?y
Enter 1st number :40
Enter 2nd number :5
Enter operator(+,-,*,/):/
result:8.0
do you want to continue(y/n) ?n


# Program 27
# Write a Python Program to Display Fibonacci Sequence Using Recursion.


In [ ]:
def fib(n):
  if n<=1:
    return n
  else:
    return fib(n-1)+fib(n-2)
for i in range(8):
  print(fib(i))

0
1
1
2
3
5
8
13


# Program 28
# Write a Python Program to Find Factorial of Number Using Recursion.

In [ ]:
def fact(n):
  if n==0 or n==1:
    return n
  else:
    return n*fact(n-1)
fact(5)

120

# Program 29 Write a Python Program to calculate your Body Mass Index.

In [ ]:
def bodymassindex(h,w):
  return round((w/h**2),2)
h=float(input('Enter height '))
w=float(input('Enter weight '))
print('welcome BMI calculator ')
bmi=bodymassindex(h,w)
print('your bmi is',bmi)
if bmi <=18.5:
  print('you are underweight ')
elif 18.5<bmi<=24.9:
  print('your wight is normal ')
else:
  print('you are obese')

Enter height 18
Enter weight 70
welcome BMI calculator 
your bmi is 0.22
you are underweight 


# Program 30 Write a Python Program to calculate the natural logarithm of any number.


In [ ]:
import math
num=float(input('Enter number '))
if num<=0:
  print('please enter a positive number ')
else:
  result=math.log(num)
  print(result)

Enter number 1.4
0.3364722366212129


# Program 31
# Write a Python Program for cube sum of first n natural numbers?

In [ ]:
def natural(n):
  s=0
  for i in range(n+1):
    s=s+(i**3)
  return s
natural(7)

784

# Program 32
#Write a Python Program to find sum of array.


In [ ]:
def soa(arr):
  total=0
  for i in arr:
    total+=i
  return total
arr=[1,2,3]
soa(arr)

6

# Program 33
# Write a Python Program to find largest element in an array

In [ ]:
def find_largest(arr):
  max_val=arr[0]
  for i in arr:
    max_val=i if i>max_val else max_val
  return max_val
arr=[67,78,89]
find_largest(arr)

89

# Program 34
# Write a Python Program for array rotation

In [ ]:
n = [1,2,3,4,5]
rotate = 2
rotate = rotate % len(n)
ans = n[rotate:] + n[:rotate]
print(ans)

[3, 4, 5, 1, 2]


# Program 35
# Write a Python Program to Split the array and add the first part to the end?

In [ ]:
arr=[1,2,3,4,5]
k=3
new_arr=arr[k:]+arr[:k]
new_arr

[4, 5, 1, 2, 3]

# Program 36
# Write a Python Program to check if given array is Monotonic

In [ ]:
def is_monotonic(arr):
  increasing=decreasing=True
  for i in range(1,len(arr)):
    if arr[i]>arr[i-1]:
      decreasing=False
    elif arr[i]<arr[i-1]:
      increasing=False
  return increasing or decreasing
array1 = [1, 2, 3, 4, 5]
array2 = [5, 4, 3, 2, 1]
array3 = [1, 3, 2, 5, 7]
print(is_monotonic(array1))
print(is_monotonic(array2))
print(is_monotonic(array3))

True
True
False


# Program 37
# Write a Python Program to Add Two Matrices.


In [ ]:
def add_matrices(matrix1,matrix2):
  result=[]
  for i in range(len(matrix1)):
    row=[]
    for j in range(len(matrix1[0])):
      row.append(matrix1[i][j]+matrix2[i][j])
    result.append(row)
  return result

matrix1 = [[1, 2, 3], [4, 5, 6]]
matrix2 = [[7, 8, 9], [10, 11, 12]]
print(add_matrices(matrix1, matrix2))


[[8, 10, 12], [14, 16, 18]]


# Program 38
# Write a Python Program to Multiply Two Matrices.


In [ ]:
def multiply_matrices(matrix1,matrix2):
  result=[[0 for j in range(len(matrix1[0]))] for i in range(len(matrix1))]
  for i in range(len(matrix1)):
    for j in range(len(matrix2[0])):
      for k in range(len(matrix2)):
        result[i][j]+=matrix1[i][k]*matrix2[k][j]
  return result
matrix1 = [[1, 2], [3,4]]
matrix2 = [[5,6], [7,8]]
print(multiply_matrices(matrix1, matrix2))

[[19, 22], [43, 50]]


# Program 39
# Write a Python Program to Transpose a Matrix.

In [ ]:
def transpose(matrix):
  result=[[0 for i in range(len(matrix))]for j in range(len(matrix[0]))]
  for i in range(len(matrix)):
    for j in range(len(matrix[0])):
      result[j][i]=matrix[i][j]
  return result
matrix = [[1, 2, 3], [4, 5, 6]]
print(transpose(matrix))

[[1, 4], [2, 5], [3, 6]]


# Program 40
# Write a Python Program to Sort Words in Alphabetic Order.


In [ ]:
def sort_words(words):
  for i in range(len(words)):
    for j in range(i+1,len(words)):
      if words[i]>words[j]:
        temp=words[i]
        words[i]=words[j]
        words[j]=temp
  return words
words = ["red", "blue", "green", "yellow", "orange", "purple",'z']
print(sort_words(words))

['blue', 'green', 'orange', 'purple', 'red', 'yellow', 'z']


# Program 41
# Write a Python Program to Remove Punctuation From a String.

In [ ]:
punctuations = '''!()-[]{};:'"\,<>./?@#$%^&*_~'''
def remove_punctuation(string):
  no_punct=''
  for chr in string:
    if chr not in punctuations:
      no_punct+=chr
  return no_punct
my_str='hi@python!'
remove_punctuation(my_str)

'hipython'

# Program 42
#list to string

In [ ]:
l=['i','am','munai']
l=''.join(l)
l

'iammunai'

# Program 43
# Write a Python program to check if the given number is a Disarium Number.

In [ ]:
def is_disarium(num):
  num_str=str(num)
  len_str=len(num_str)
  sum=0
  for i in range(len(num_str)):
    sum+=int(num_str[i])**(i+1)
  return num==sum
is_disarium(89)

True

## Program 44
# Write a Python program to print all disarium numbers between 1 to 100.

In [ ]:
def is_disarium(num):
  num_str = str(num)
  len_str = len(num_str)
  sum = 0
  for i in range(len_str):
    sum += int(num_str[i]) ** (i + 1)
  return num == sum
disarium_numbers = [num for num in range(1, 101) if is_disarium(num)]
for num in disarium_numbers:
 print(num)


1
2
3
4
5
6
7
8
9
89


# Program 45
# Write a Python program to check if the given number is Happy Number

In [ ]:
def is_happy_number(num):
  seen=set()
  while num!=1 and num not in seen:
    seen.add(num)
    num=sum(int(i)**2 for i in str(num))
  return num==1
is_happy_number(23)

True

# Program 46
# Write a Python program to print all happy numbers between 1 and 100.


In [ ]:
def is_happy_number(num):
  seen=set()
  while num!=1 and num not in seen:
    seen.add(num)
    num=sum(int(i)**2 for i in str(num))
  return num==1
happy_numbers=[]
for num in range(1,101):
  if is_happy_number(num):
    happy_numbers.append(num)
print(happy_numbers)


[1, 7, 10, 13, 19, 23, 28, 31, 32, 44, 49, 68, 70, 79, 82, 86, 91, 94, 97, 100]


# Program 47
# Write a Python program to determine whether the given number is a Harshad Number.


In [ ]:
def is_harshad(num):
  digit_sum=sum(int(i) for i in str(num))
  return num%digit_sum==0
is_harshad(42)

True

# Program 48
# Write a Python program to print all pronic numbers between 1 and 100.

In [ ]:
def is_pronic(num):
  for n in range(1,int(num**0.5)+1):
    if n*(n+1)==num:
      return True
  return False
for i in range(1,101):
  if is_pronic(i):
    print(i)

2
6
12
20
30
42
56
72
90


# Program 49 Write a Python program to sum all numbers in the list.


In [ ]:
numbers=[7,9,2]
sum=0
for i in (numbers):
  sum+=i
print(sum)

18


# Program 50
# Write a Python program to Multiply all numbers in the list.


In [ ]:
num=[9,3,5]
mul=1
for i in num:
  mul*=i
print(mul)

135


# Program 51
# Write a Python program to find smallest number in a list.


In [ ]:
num=[6,-8,8,9]
min=num[0]
for i in num:
  if i<min:
    min=i
print(min)

-8


# Program 52
# Write a Python program to find largest number in a list.

In [ ]:
num=[8,9,90]
max=num[0]
for i in num:
  if i>max:
    max=i
print(max)

90


# Program 53
# Write a Python program to find second largest number in a list

In [ ]:
def second_high(num):
  num=list(set(num))
  num.sort()
  return num[-2]
nums=[6,7,9,90]
second_high(nums)

9

#alternative

In [ ]:
def sec_max(arr):
  sublist=[x for x in arr if x<max(arr)]
  return max(sublist)
list=[8,9,0,7]
sec_max(list)

8

# Program 54
# Write a Python program to find N largest elements from a list.

In [ ]:
def n_largest(list1,n):
  list1.sort(reverse=True)
  return list1[:n]
list1=[89,78,66,56]
n=3
print(n_largest(list1,n))

[89, 78, 66]


# Program 55
# Write a Python program to print even numbers in a list.


In [ ]:
numbers = [67, 690, 56]
result = [num for num in numbers if num % 2 == 0]
print(result)

[690, 56]


# Program 56
# Write a Python program to print odd numbers in a List.


In [ ]:
numbers=[9,6,7,49,37]
result=[num for num in numbers if num%2!=0]
print(result)

[9, 7, 49, 37]


# Program 57
# Write a Python program to Remove empty List from List.

In [ ]:
list_of_lists=[[1,2,3],[],[4,5],[],[6,7,8],[]]
filtered_list=[ i for i in list_of_lists if i]
print(filtered_list)

[[1, 2, 3], [4, 5], [6, 7, 8]]


# Program 58 Write a Python program to Cloning or Copying a list.

In [ ]:
original_list=[1,2,3,4,5]
cloned_list=original_list[:]
print(cloned_list)

[1, 2, 3, 4, 5]


# Program 59
# Write a Python program to Count occurrences of an element in a list.

In [ ]:
def count_occurrences(lst,element):
  count=0
  for item in lst:
    if item==element:
      count+=1
  return count
my_list=[90,7,7,8,8,8]
ele_count=8
result=count_occurrences(my_list,ele_count)
result

3

# Program 60
# Write a Python program to find words which are greater than given length k.

In [ ]:
def words_greater_than_k(sentence,k):
  words=sentence.split()
  result=[word for word in words if len(word)>k]
  return result
input_sentence = "This is a sample sentence to demonstrate the program"
length_k = 5
words = words_greater_than_k(input_sentence, length_k)
words

['sample', 'sentence', 'demonstrate', 'program']

# Program 61
# Write a Python program for removing 𝑖^th character from a string.

In [ ]:
def remove_ith_character(input_string,i):
  if i<=0 or i>=len(input_string):
    return 'invalid index'
  else:
    return input_string[:i]+input_string[i+1:]
string='example'
index_to_remove=2
result=remove_ith_character(string,index_to_remove)
result

'exmple'

# **Program** 62
# Write a Python program to split and join a string

In [ ]:
def split_and_join(input_string):
  word_list=input_string.split()
  joined_string='_'.join(word_list)
  return word_list,joined_string
string='this is sample string'
split_str,join_str=split_and_join(string)
print(split_str)
print(join_str)

['this', 'is', 'sample', 'string']
this_is_sample_string


# Program 63
# Write a Python program to check if a given string is binary string or not

In [ ]:
def is_binary_string(input_string):
  return set(input_string).issubset({'0','1'})
is_binary_string('101')


True

# Program 64
# Write a Python program to find uncommon words from two Strings.


In [ ]:
def find_uncommon_words(str1,str2):
  return set(str1.split())^(set(str2.split()))
string_1 = "hi how are u?"
string_2 = "hi how was she?"
find_uncommon_words(string_1,string_2)



{'are', 'she?', 'u?', 'was'}

#Program 65
#Write a Python program to find all duplicate characters in string.

In [ ]:
def duplicate_chars(input_string):
  duplicate=[]
  seen=set()
  for char in input_string:
    if char in seen:
      duplicate.append(char)
    else:
      seen.add(char)
  return list(set(duplicate))
duplicate_chars("piyush sharma")

['h', 's', 'a']

# Program 66
# Write a Python Program to check if a string contains any special character.


In [ ]:
import re
def contains_special_character(input_string):
  regex=re.compile('[@_!#$%^&*()<>?/\|}{~:]')
  if regex.search(input_string):
    return True
  else:
    return False
contains_special_character('hello!@%')


True

# Program 67
# Write a Python program to Extract Unique dictionary values.


In [ ]:
def extract_values(input_dict):
  unique_val=list(set(input_dict.values()))
  return unique_val
sample_dict={'a':1,'b':2,'c':1,'d':3,'e':2}
unique_val=extract_values(sample_dict)
print(unique_val)

[1, 2, 3]


# Program 68
# Write a Python program to find the sum of all items in a dictionary.


In [ ]:
def sum_dict_values(input_dict):
  total_sum=0
  for value in input_dict.values():
    total_sum+=value
  return total_sum
def main():
  sample_dict={'a':10,'b':20,'c':30,'d':40}
  result=sum_dict_values(sample_dict)
  print(result)
main()

100


# Program 69
# Write a Python program to Merging two Dictionaries.

In [ ]:
def merge_dict(d1,d2):
  merged_dict=d1.copy()
  merged_dict.update(d2)
  return merged_dict
dict_a={'a':1,"b":2}
dict_b={'c':3,'d':4}
result=merge_dict(dict_a,dict_b)
print(result)

{'a': 1, 'b': 2, 'c': 3, 'd': 4}


#Program 70
# Write a Python program to convert key-values list to flat dictionary.

In [ ]:
def convert_to_flat_dict(key_value_list):
  return dict(key_value_list)
key_value_list=[('a',1),('b',2),('c',3),('d',4)]
result=convert_to_flat_dict(key_value_list)
print(result)

{'a': 1, 'b': 2, 'c': 3, 'd': 4}


# Program 71
# Write a Python program to insertion at the beginning in OrderedDict.


In [ ]:
from collections import OrderedDict
ordered_dict=OrderedDict([('b',2),('c',3),('d',4)])
new_item=('a',1)
new_ordered_dict=OrderedDict([new_item])
new_ordered_dict.update(list(ordered_dict.items()))
print(new_ordered_dict)

OrderedDict([('a', 1), ('b', 2), ('c', 3), ('d', 4)])


# 72  Write a Python program to check order of character in string using OrderedDict().


# Program 131
# Create a function that returns the mean of all digits.
# Examples
# mean(42) ➞ 3
# mean(12345) ➞ 3
# mean(666) ➞ 6
# Notes
# The mean of all digits is the sum of digits / how many digits there are (e.g. mean
# of digits in 512 is (5+1+2)/3(number of digits) = 8/3=2).
# The mean will always be an integer.


In [58]:
def mean(n):
  n_str=str(n)
  digit_sum=sum(int(digit) for digit in n_str)
  digit_count=len(n_str)
  return int(digit_sum/digit_count)
mean(666)



6

# Program 132
# Create a function that takes an integer and returns a list from 1 to the given number,
# where:
# 1. If the number can be divided evenly by 4, amplify it by 10 (i.e. return 10 times the
# number).
# 2. If the number cannot be divided evenly by 4, simply return the number.
# Examples

#amplify(4) ➞ [1, 2, 3, 40]
#amplify(3) ➞ [1, 2, 3]
#amplify(25) ➞ [1, 2, 3, 40, 5, 6, 7, 80, 9, 10, 11, 120, 13, 14, 15, 160, 17, 18, 19, 200, 21,
#22, 23, 240, 25]
#Notes
#The given integer will always be equal to or greater than 1.
#Include the number (see example above).
#To perform this problem with its intended purpose, try doing it with list
# comprehensions. If that's too difficult, just solve the challenge any way you can.


In [52]:
def amplify(num):
  return [n*10 if n%4==0 else n for n in range(1,num+1)]
amplify(25)

[1,
 2,
 3,
 40,
 5,
 6,
 7,
 80,
 9,
 10,
 11,
 120,
 13,
 14,
 15,
 160,
 17,
 18,
 19,
 200,
 21,
 22,
 23,
 240,
 25]

# Program 133
# Create a function that takes a list of numbers and return the number that's unique.
# Examples
# unique([3, 3, 3, 7, 3, 3]) ➞ 7
# unique([0, 0, 0.77, 0, 0]) ➞ 0.77
# unique([0, 1, 1, 1, 1, 1, 1, 1]) ➞ 0
# Notes
# Test cases will always have exactly one unique number while all others are the same.

In [47]:
def unique(numbers):
  digit_count={}
  for number in numbers:
    if number in digit_count:
      digit_count[number]+=1
    else:
      digit_count[number]=1
  for num,count in digit_count.items():
    if count==1:
      return num
unique([0, 1, 1, 1, 1, 1, 1, 1])

0

## Program 134
# Your task is to create a Circle constructor that creates a circle with a radius provided
# by an argument. The circles constructed must have two getters getArea() (PIr^2) and
# getPerimeter() (2PI*r) which give both respective areas and perimeter
# (circumference).
# For help with this class, I have provided you with a Rectangle constructor which you
# can use as a base example.
# Examples
# circy = Circle(11)
# circy.getArea()
#Should return 380.132711084365
#circy = Circle(4.44)
#circy.getPerimeter()
#Should return 27.897342763877365
#Notes
#Round results up to the nearest integer.


In [39]:
import math
class circle:
  def __init__(self,radius):
    self.radius=radius
  def getarea(self):
    return (math.pi*self.radius**2)
  def getperimeter(self):
    return (2*math.pi*self.radius)
c=circle(4.44)
c.getarea()
c.getperimeter()


27.897342763877365

# Program 135
# Create a function that takes a list of strings and return a list, sorted from shortest to
# longest.
# Examples
# sort_by_length(["Google", "Apple", "Microsoft"]) ➞ ["Apple", "Google", "Microsoft"]
# sort_by_length(["Leonardo", "Michelangelo", "Raphael", "Donatello"]) ➞ ["Raphael",
# "Leonardo", "Donatello", "Michelangelo"]
# sort_by_length(["Turing", "Einstein", "Jung"]) ➞ ["Jung", "Turing", "Einstein"]
# Notes


In [34]:
def sort_by_length(strings):
  return sorted(strings,key=len)
sort_by_length(["Turing", "Einstein", "Jung"])

['Jung', 'Turing', 'Einstein']

# Program 136
# Create a function that validates whether three given integers form a Pythagorean
# triplet. The sum of the squares of the two smallest integers must equal the square of
# the largest number to be validated.
# Examples
# is_triplet(3, 4, 5) ➞ True
# 3² + 4² = 25
# 5² = 25
# is_triplet(13, 5, 12) ➞ True
# 5² + 12² = 169
# 13² = 169
# is_triplet(1, 2, 3) ➞ False
# 1² + 2² = 5
# 3² = 9
# Notes
# Numbers may not be given in a sorted order.


In [31]:
def is_trip(a,b,c):
  num=[a,b,c]
  num.sort()
  return num[0]**2+num[1]**2==num[2]**2
is_trip(13,5,12)

True

# Program 137
# Create a function that takes three integer arguments (a, b, c) and returns the amount
# of integers which are of equal value.
# Examples
# equal(3, 4, 3) ➞ 2
# equal(1, 1, 1) ➞ 3
# equal(3, 4, 1) ➞ 0
# Notes
# Your function must return 0, 2 or 3.


In [29]:
def equal(a,b,c):
  if a==b==c:
    return 3
  elif a==b or a==c:
    return 2
  else:
    return 0
equal(3,4,3)

2

# Program 138
# Write a function that converts a dictionary into a list of keys-values tuples.
# Examples
# dict_to_list({
# "D": 1,
# "B": 2
# "C": 3
# }) ➞ [("B", 2), ("C", 3), ("D", 1)]
# dict_to_list({
# "likes": 2,
# "dislikes": 3,
# "followers": 10
# }) ➞ [("dislikes", 3), ("followers", 10), ("likes", 2)]
# Notes
# Return the elements in the list in alphabetical order

In [25]:
def dict_to_list(input_dict):
  sorted_dict=sorted(input_dict.items())
  result=[(key,value)for key,value in sorted_dict]
  return result
dict_to_list({
"likes": 2,
"dislikes": 3,
"followers": 10
})

[('dislikes', 3), ('followers', 10), ('likes', 2)]

# Program 139
# Write a function that creates a dictionary with each (key, value) pair being the (lower
# case, upper case) versions of a letter, respectively.
# Examples
# mapping(["p", "s"]) ➞ { "p": "P", "s": "S" }
# mapping(["a", "b", "c"]) ➞ { "a": "A", "b": "B", "c": "C" }
# mapping(["a", "v", "y", "z"]) ➞ { "a": "A", "v": "V", "y": "Y", "z": "Z" }
# Notes
# All of the letters in the input list will always be lowercase.


In [18]:
def mapping(letters):
  res={}
  for letter in letters:
    res[letter]=letter.upper()
  return res
mapping(["a", "v", "y", "z"])

{'a': 'A', 'v': 'V', 'y': 'Y', 'z': 'Z'}

# Program 140
# Write a function, that replaces all vowels in a string with a specified vowel.
# Examples
# vow_replace("apples and bananas", "u") ➞ "upplus und bununus"
# vow_replace("cheese casserole", "o") ➞ "chooso cossorolo"
# vow_replace("stuffed jalapeno poppers", "e") ➞ "steffed jelepene peppers"
# Notes
# All words will be lowercase. Y is not considered a vowel.

In [10]:
def vow_replace(input_str,vowel):
  result=''
  vowels='aeiou'
  for char in input_str:
    if char in vowels:
      result+=vowel
    else:
      result+=char
  return result
vow_replace("stuffed jalapeno poppers", "e")

'steffed jelepene peppers'

# Program 141
# Create a function that takes a string as input and capitalizes a letter if its ASCII code
# is even and returns its lower case version if its ASCII code is odd.
# Examples
# ascii_capitalize("to be or not to be!") ➞ "To Be oR NoT To Be!"
# ascii_capitalize("THE LITTLE MERMAID") ➞ "THe LiTTLe meRmaiD"
# ascii_capitalize("Oh what a beautiful morning.") ➞ "oH wHaT a BeauTiFuL
# moRNiNg."

In [9]:
def asci_capitalize(input_str):
  result=""
  for char in input_str:
    if ord(char)%2==0:
      result+=char.upper()
    else:
      result+=char.lower()
  return result
asci_capitalize("Oh what a beautiful morning.")

'oH wHaT a BeauTiFuL moRNiNg.'